Source/Core/DolphinNoGUI/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+if(APPLE)
+	set(DOLPHIN_EXE_BASE Dolphin)
+else()
+	set(DOLPHIN_EXE_BASE dolphin-emu)
+endif()
+
+set(NOGUI_SRCS MainNoGUI.cpp)
+
+list(APPEND LIBS core uicommon cpp-optparse)
+
+if(USE_X11 OR ENABLE_HEADLESS)
+	set(DOLPHIN_NOGUI_EXE ${DOLPHIN_EXE_BASE}-nogui)
+	add_executable(${DOLPHIN_NOGUI_EXE} ${NOGUI_SRCS})
```

Comment:
Don't do it like this please.
A target should have the same name all the time.

If you really want to control the output name, use the property [OUTPUT_NAME](https://cmake.org/cmake/help/v3.7/prop_tgt/OUTPUT_NAME.html). I don't believe we need that though, dolphin-nogui doesn't really work on anything than Linux right now, so it could just have that name.